### PR TITLE
[release-4.10] Bug 2088040: Delete ChassisPrivate along with Chassis

### PR DESCRIPTION
**Pull request by jcaamano**
CARRY of https://github.com/openshift/ovn-kubernetes/pull/1064/commits/07b3de6cbc7a94103ad5c176764abddd65dd8979 && https://github.com/openshift/ovn-kubernetes/pull/1064/commits/326337d39d6b179b8a1592e845afc099c9708343

Fixed some broken test that would not even compile.

**- What this PR does and why is it needed**

**- Special notes for reviewers**

**- How to verify it**

**- Description for the changelog**

### CODE REVIEW

File: go-controller/pkg/ovn/master.go [hunk 1]
```diff
@@ -1341,22 +1348,22 @@ func (oc *Controller) syncNodesPeriodic() {
...
+		chassisHostNameMap[chassis.Hostname] = chassis
 	}
 
 	//delete existing nodes from the chassis map.
 	for _, nodeName := range nodeNames {
 		delete(chassisHostNameMap, nodeName)
 	}
 
-	staleChassisNames := []string{}
+	staleChassis := []*sbdb.Chassis{}
```
Comment (flavio-fernandes) [inline go-controller/pkg/ovn/master.go:1361]: this is a bit different on 4.11 now, but the backport looks right

**Review by flavio-fernandes — COMMENTED** [review]
/lgtm

File: go-controller/pkg/ovn/master.go (hunk 1 above)
Comment (jcaamano) [reply go-controller/pkg/ovn/master.go:1361]: yeah, in 4.11 we replaced all arrays of model structs to arrays of models (pointers to structs). I did not want it mixed in 4.10 so I used the 4.10 form.

**Comment by asood-rh** [comment]
/label cherry-pick-approved

The build with PR tested details in https://bugzilla.redhat.com/show_bug.cgi?id=2088040#c1

**Pull request merged**